plugins/modules/provision_workflow_manager.py

```diff
@@ -590,43 +592,36 @@ def get_site_type(self, site_name_hierarchy=None):
 
     def get_site_details(self, site_name_hierarchy=None):
         """
-        Fetches the id and existance of the site
-
         Parameters:
-          - self: The instance of the class containing the 'config' attribute
-                  to be validated.
-          - site_name_hierarchy: Name of the site collected from the input.
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
         Returns:
-          - site_id: A string indicating the id of the site.
-          - site_exits: A boolean value indicating the existance of the site.
-        Example:
-          Post creation of the validated input, this method gets the
-          id of the site.
+            tuple: A tuple containing two values:
+            - site_exists (bool): A boolean indicating whether the site exists (True) or not (False).
+            - site_id (str or None): The ID of the site if it exists, or None if the site is not found.
+        Description:
+            This method checks the existence of a site in the Catalyst Center. If the site is found,it sets 'site_exists' to True,
+            retrieves the site's ID, and returns both values in a tuple. If the site does not exist, 'site_exists' is set
+            to False, and 'site_id' is None. If an exception occurs during the site lookup, an exception is raised.
         """
 
         site_exists = False
         site_id = None
-        try:
-            response = self.dnac_apply['exec'](
-                family="sites",
-                function='get_site',
-                params={"name": site_name_hierarchy},
-                op_modifies=True
-            )
-        except Exception:
-            self.log("Exception occurred as \
-                site '{0}' was not found".format(self.want.get("site_name")), "CRITICAL")
-            self.module.fail_json(msg="Site not found", response=[])
+        response = None
 
-        if response:
-            self.log("Received site details\
-                for '{0}': {1}".format(site_name_hierarchy, str(response)), "DEBUG")
+        try:
+            response = self.get_site(site_name_hierarchy)
```

Comment:
if you want to get site_id, then use get_site_id API instead of get_site..

Reply:
updated the code